analyses/npl/08_water_level_events.md

```diff
@@ -0,0 +1,429 @@
+to make sure we're comparing similar types of events
+
+```python
+for station in STATIONS:
+    print(station)
+    df_station = df_station_dict[station]
+    df_station = df_station.groupby(df_station.index.year).sum() > 0
+    print(f"Total years: {len(df_station)}")
+    for level_type in LEVEL_TYPES:
+        n = df_station[f"event_{level_type}"].sum()
+        print(f"{level_type} level: {n}")
+    for rp in RP_LIST:
+        n = df_station[f"event_rp{rp}"].sum()
+        print(f"1 in {rp} y: {n}")
+    print('\n')
+```
+
+Unfortuantely there is some mismatch. 
+For Chatara, it seems that warning level corresponds to 1 in 1.5 year, and danger level to 1 in 2 year.
+For chisapani, warning level is closer to 1 in 2 year (probably more like 1 in 3 year) and danger level to 1 in 5 year. 
+
+```python
+
+# Settle on RP and level type
+rp = MAIN_RP
+event_level_type = 'danger'
+
+days_before_buffer = 3 # How many days the true event can occur before the GloFAS event
+days_after_buffer = 30 # How many days the tru event can occur after the GloFAS event
+
+df_station_stats = pd.DataFrame(columns=['station', 'TP', 'FP', 'FN', 'wl_days'])
+
+for station in STATIONS:
+    df_station = df_station_dict[station]
+    for wl_days in [None] + WL_DAYS:
+        df_true_events = df_station[df_station[f"event_{event_level_type}"]]
+        df_true_events['detections'] = 0
+        TP = 0
+        FP = 0
+        if wl_days is None:
+            glofas_cname = f"event_rp{rp}"
+        else:
+            glofas_cname = f"event_rp{rp}_wl{wl_days}"
+        glofas_events = df_station[df_station[glofas_cname]].index
+        if wl_days is None:
+            print(f"{station}")
+            print(f"True events: {len(df_true_events)}")
+            print(f"Glofas events: {len(glofas_events)}")
+        for glofas_event in glofas_events:
+            # Check if any events are around that date
+            days_offset = (df_true_events.index - glofas_event) /  np.timedelta64(1, 'D')
+            detected = (days_offset >= -1 * days_before_buffer) & (days_offset <= days_after_buffer)
+            df_true_events.loc[detected, 'detections'] += 1
+            # If there were any detections, it's  a TP. Otherwise a FP
+            if sum(detected):
+                TP += 1
+            else:
+                FP += 1
+        df_station_stats = df_station_stats.append({
+            'station': station,
+            'TP': TP,
+            'FP': FP,
+            'FN': len(df_true_events[df_true_events['detections'] == 0]),
+            'wl_days': wl_days
+        }, ignore_index=True)
+df_station_stats[df_station_stats['wl_days'].isnull()]
+```
+
+# 
+
+
+## Make plots for presentation
+
+```python
+def plot_arrow(ax, x, y, c):
+    ax.annotate(" ", 
+                     xy=(x, y+y*0.3),
+                     xytext=(x, y+y*0.5),
+                    arrowprops=dict(facecolor=c, shrink=0.05, headlength=3,
+                               width=1, headwidth=3, lw=0.5))
```

Comment:
Maybe nice to add a bit of transparency to these arrows to better see when they're overlapping.

Reply:
Good idea!